sources/web/datalab/static/datalab.js

```diff
@@ -12,13 +12,17 @@
  * the License.
  */
 
+var debug = {
+  enabled: true,
+  log: function() { console.log.apply(console, arguments); }
+};
+
+function shimWebSockets() {
+    return location.host.substr(-12) === '.appspot.com';
```

Comment:
websockets didn't work with SauceLabs, but we changed that. I just tested that it works with headless Firefox, so we're fine.